packages/nuxt/src/app/nuxt.ts

```diff
@@ -252,12 +252,12 @@ export function createNuxtApp (options: CreateOptions) {
       get nuxt () { return __NUXT_VERSION__ },
       get vue () { return nuxtApp.vueApp.version },
     },
-    payload: reactive({
-      data: {},
-      state: {},
+    payload: shallowReactive({
+      data: shallowReactive({}),
+      state: reactive({}),
       once: new Set<string>(),
-      _errors: {},
-      ...(import.meta.client ? window.__NUXT__ ?? {} : { serverRendered: true }),
+      _errors: shallowReactive({}),
+      error: undefined,
```

Comment:
```suggestion
```
Is it needed ? This is not going to be serialized if undefined

Reply:
Good spot. I think I added it at some point to ensure that `useError` continued to work - but that was when I was experimenting with not even marking the payload as `shallowReactive`.

https://github.com/nuxt/nuxt/blob/main/packages/nuxt/src/app/composables/error.ts#L10